File: c++/509 - Fibonacci Number.cpp
// 509
// Fibonacci Number
// https://leetcode.com/problems/fibonacci-number/
// c++
// easy
// O(n)
// O(1)
// basic

class Solution {
public:
    int fib(int N) {
        int a=0;
        int b=1;
        for (int i=0; i < N; i++) {
            b=a+b;
            a=b-a;
        }
    
        return a;
    }
};

File: python/509 - Fibonacci Number.py
# 509
# Fibonacci Number
# https://leetcode.com/problems/fibonacci-number/
# python
# easy
# O(n)
# O(1)
# basic

class Solution:
    def fib(self, N: int) -> int:
        a=0
        b=1
        for _ in range(N):
            b=a+b
            a=b-a
        return a
